presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -413,6 +411,7 @@ public boolean getMoreResults(int current)
     {
         checkOpen();
 
+        currentUpdateCount.set(-1);
```

Comment:
I think we should do this inside the if block for `CLOSE_CURRENT_RESULT`. If this method is called with `KEEP_CURRENT_RESULT`, we throw an exception and don't close the result set, so clearing the update count would be inconsistent.

Reply:
Let's also clear the update type here (a Presto extension):
```java
currentUpdateType.set(null);
```

Reply:
Thanks, I see what you mean. I'll add the `currentUpdateType` call and merge this.